runcoverage.sh

```diff
@@ -23,40 +23,67 @@ then
   fi
 fi
 
-$NUGET install -Verbosity quiet -OutputDirectory packages -Version 4.6.519 OpenCover
+$NUGET install -Verbosity quiet -OutputDirectory packages -Version 2017.1.20170613.162720 JetBrains.dotCover.CommandLineTools
 $NUGET install -Verbosity quiet -OutputDirectory packages -Version 2.4.5.0 ReportGenerator
-OPENCOVER=$PWD/packages/OpenCover.4.6.519/tools/OpenCover.Console.exe
+
+DOTCOVER=$PWD/packages/JetBrains.dotCover.CommandLineTools.2017.1.20170613.162720/tools/dotCover.exe
 REPORTGENERATOR=$PWD/packages/ReportGenerator.2.4.5.0/tools/ReportGenerator.exe
 FIND=/usr/bin/find
 
-rm -rf coverage
-mkdir coverage
+if [ ! -d coverage ]
+then
+ mkdir coverage
+fi
 
 while read csproj
 do
   testdir=$(dirname $csproj)
   api=$(basename $(realpath $testdir/..))
+
+  [ ! -d "$testdir/bin/Release/net452" ] && [ ! -d "$testdir/bin/Release/netcoreapp1.0" ] && continue
+
   echo "Running coverage for $api"
-  # OpenCover is picky about how it finds the excluded files. There may be a better approach than this,
```

Comment:
Looking at https://www.jetbrains.com/help/dotcover/dotCover__Console_Runner_Commands.html#cover, there is filtering available, but it looks like it's based on built artifacts (type, member, module) rather than source. Hmm. I'll investigate this.